algorithms/linfa-elasticnet/src/algorithm.rs

```diff
@@ -56,6 +61,64 @@ where
     }
 }
 
+// impl<F, D, T> Fit<ArrayBase<D, Ix2>, T, ElasticNetError> for ElasticNetValidParams<F>
```

Comment:
@YuhanLiin If I uncomment this line, my compiler says that I'm redefining the `Fit` trait for `ElasticNetValidParams`.

Reply:
Best way to resolve this is to change the `Fit` impl for single-task ElasticNet to `impl<F, D> Fit<ArrayBase<D, Ix2>, ArrayBase<D, Ix1>, ElasticNetError> for ElasticNetValidParams<F>`. This avoids conflicting between the two `Fit` impls on `ElasticNetValidParams`. The original `Fit` impl is only used for 1D outputs anyways so this change won't break anything. This way you won't need a multivariable version of the param struct.

Reply:
If you want to stick with having 2 param types you can do something like:
```rust
pub struct ElasticNetParamsBase<const MULTI: bool> {
  ...
}

pub type ElasticNetParams = ElasticNetParamsBase<false>;
pub type MultiElasticNetParams = ElasticNetParamsBase<true>;
```
Then just put all the impls on `ElasticNetParamsBase` to avoid repeating everything.